qcodes/tests/drivers/test_ami430_visa.py

```diff
@@ -486,6 +487,170 @@ def test_ramp_rate_exception(current_driver):
         ix.ramp_rate(target_ramp_rate)
 
 
+def test_simultaneous_ramp_mode_does_not_reset_individual_axis_ramp_rates_if_nonblocking_ramp(
+    current_driver, caplog, request
+):
+    ami3d = current_driver
+
+    ami3d.cartesian((0.0, 0.0, 0.0))
+
+    restore_parameters_stack = ExitStack()
+    request.addfinalizer(restore_parameters_stack.close)
+
+    restore_parameters_stack.callback(ami3d.cartesian, (0.0, 0.0, 0.0))
+
+    restore_parameters_stack.enter_context(
+        ami3d._instrument_x.ramp_rate.restore_at_exit()
+    )
+    restore_parameters_stack.enter_context(
+        ami3d._instrument_y.ramp_rate.restore_at_exit()
+    )
+    restore_parameters_stack.enter_context(
+        ami3d._instrument_z.ramp_rate.restore_at_exit()
+    )
+
+    restore_parameters_stack.enter_context(ami3d.ramp_mode.set_to("simultaneous"))
+
+    restore_parameters_stack.enter_context(ami3d.block_during_ramp.set_to(False))
+
```

Comment:
All of the above is to reset the instrument to a known good state after the test right ?

Would it make sense to refactor that out into a context manager

Reply:
most of the above is indeed to reset the instrument, but some of the statements are part of "setting up the instrument into a state required for the test". the ask is fair, but not super critical. (ideally, we'd restore all parameters of the instrument at the teardown of the fixture)